source/install/upgrade_from_mongodb.rst

```diff
@@ -100,5 +117,35 @@ Upgrading on Red Hat Enterprise Linux or CentOS
    example, existing data may not be compatible with the default WiredTiger
    storage engine.
 
+Upgrading a replica set
+================================================================================
+
+The :term:`rolling restart <Rolling restart>` method lets you upgrade your replica set to |PSMDB| with minimum downtime. You upgrade the nodes one by one while the whole cluster remains operational.   
+
+Upgrade the replica set secondary nodes  
+--------------------------------------------------------------------------------
+
+1. Upgrade a secondary node in a replica set as described in :ref:`upgrade_standalone`. Use the instructions relevant to your operating system. 
+#. Wait for the node to rejoin with the replica set members and report the SECONDARY status. 
+#. Repeat the upgrade procedure on the remaining secondary nodes. 
+   
+Upgrade the primary node
+--------------------------------------------------------------------------------
+
+1. Step down the primary node: :command:`rs.stepDown()`.
+2. Wait for the remaining nodes to elect a new primary. Run :command:`rs.status()` to verify that the former primary node reports as SECONDARY.
+#. Upgrade the node as described in :ref:`upgrade_standalone`.
```

Comment:
```suggestion
Upgrade some but not all replica set nodes  
--------------------------------------------------------------------------------

1. Upgrade a node in a replica set as described in :ref:`upgrade_standalone`. Use the instructions relevant to your operating system. It is better to do it to the nodes that are currently secondary role so there won't be an extra election. If you do it to the node that is the current primary use :command:`rs.stepDown()` before shutting it down.
#. Wait for the node to rejoin with the replica set members, resync and report that it is in SECONDARY status. 
#. Optional: repeat the upgrade procedure on other but not all nodes.

Test the new version in the primary role before all nodes upgraded
----------------------------------------------------------------------------------------

This step is optional. Its purpose is to give a testing stage with the new version as primary whilst you still have at least one of the old version nodes. This will make rolling back a little quicker if you choose to do so.

1. Use :command:`rs.stepDown()` on the current primary to start an election and influence the primary role to switch to one of the new version nodes. If you have multiple nodes of the previous version use :command:`rs.freeze()` on them to make sure one of the new version nodes becomes primary.
 
Upgrade the last node(s)
--------------------------------------------------------------------------------

1. If any previous version nodes are the current primary node step them down: :command:`rs.stepDown()`.
2. Wait for the remaining nodes to elect a new primary. Run :command:`rs.status()` to verify that the former primary node reports as SECONDARY.
#. Upgrade the node as described in :ref:`upgrade_standalone`.
```

Hi Anastasia. I've added a middle section explaining the test-new-as-primary stage.